--- 1/src/Config.cpp
+++ 1/src/Config.cpp
@@ -1113,17 +1113,5 @@
 std::string
 Config::default_temporary_dir() const
 {
-  static const std::string run_user_tmp_dir = [] {
-#ifndef _WIN32
-    const char* const xdg_runtime_dir = getenv("XDG_RUNTIME_DIR");
-    if (xdg_runtime_dir && Stat::stat(xdg_runtime_dir).is_directory()) {
-      auto dir = FMT("{}/ccache-tmp", xdg_runtime_dir);
-      if (Util::create_dir(dir) && access(dir.c_str(), W_OK) == 0) {
-        return dir;
-      }
-    }
-#endif
-    return std::string();
-  }();
-  return !run_user_tmp_dir.empty() ? run_user_tmp_dir : m_cache_dir + "/tmp";
+  return m_cache_dir + "/tmp";
 }
