--- clucene-core-2.3.3.4/src/shared/cmake/MacroCheckGccVisibility.cmake
+++ clucene-core-2.3.3.4/src/shared/cmake/MacroCheckGccVisibility.cmake
@@ -15,7 +15,7 @@
    # get the gcc version
    exec_program(${CMAKE_C_COMPILER} ARGS --version OUTPUT_VARIABLE _gcc_version_info)
 
-   string (REGEX MATCH "[3456]\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")
+   string (REGEX MATCH "[013-9]\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")
    if (NOT _gcc_version)
    
       # clang reports: clang version 1.1 (trunk 95754)
