--- 1/lisp/info.el
+++ 1/lisp/info.el
@@ -497,6 +497,9 @@
       '( (".gz"       . "gunzip")
 	 (".z"        . "gunzip")
 	 (".bz2"      . ("bzip2" "-dc"))
+	 (".br"       . ("brotli" "-dcf"))
+	 (".lz4"      . ("lz4" "-d" "-q"))
+	 (".zst"      . ("zst" "-d" "-f" "-q" "--long=31"))
 	 (".inz"      . "gunzip")
 	 (".igz"      . "gunzip")
 	 (".info.Z"   . "gunzip")
@@ -515,12 +516,18 @@
        (".info.gz"   . "gunzip")
        (".info.z"    . "gunzip")
        (".info.bz2"  . ("bzip2" "-dc"))
+       (".info.br"   . ("brotli" "-dcf"))
+	(".info.lz4"  . ("lz4" "-d" "-q"))
+	(".info.zst"  . ("zst" "-d" "-f" "-q" "--long=31"))
        (".info.xz"   . "unxz")
        (".info"      . nil)
        ("-info.Z"    . "uncompress")
        ("-info.Y"    . "unyabba")
        ("-info.gz"   . "gunzip")
        ("-info.bz2"  . ("bzip2" "-dc"))
+       ("-info.br"   . ("brotli" "-dcf"))
+	("-info.lz4"  . ("lz4" "-d" "-q"))
+	("-info.zst"  . ("zst" "-d" "-f" "-q" "--long=31"))
        ("-info.z"    . "gunzip")
        ("-info.xz"   . "unxz")
        ("-info"      . nil)
@@ -529,6 +534,9 @@
        ("/index.gz"  . "gunzip")
        ("/index.z"   . "gunzip")
        ("/index.bz2" . ("bzip2" "-dc"))
+       ("/index.br"  . ("brotli" "-dcf"))
+	("/index.lz4" . ("lz4" "-d" "-q"))
+	("/index.zst" . ("zst" "-d" "-f" "-q" "--long=31"))
        ("/index.xz"  . "unxz")
        ("/index"     . nil)
        (".Z"         . "uncompress")
@@ -536,6 +543,9 @@
        (".gz"        . "gunzip")
        (".z"         . "gunzip")
        (".bz2"       . ("bzip2" "-dc"))
+       (".br"        . ("bro" "-dcf"))
+	(".lz4"       . ("lz4" "-d" "-q"))
+	(".zst"       . ("zst" "-d" "-f" "-q" "--long=31"))
        (".xz"        . "unxz")
        (""           . nil)))
   "List of file name suffixes and associated decoding commands.
--- 1/lisp/jka-cmpr-hook.el
+++ 1/lisp/jka-cmpr-hook.el
@@ -238,6 +238,34 @@
      "XZ compressing"     "xz"           ("-c" "-q")
      "XZ uncompressing"   "xz"           ("-c" "-q" "-d")
      t nil "\3757zXZ\0"]
+    ["\\.bro?\\'"
+     "brotli compressing" "brotli" ("-c" "-w" "24" "--best")
+     "brotli uncompressing" "brotli" ("-c" "-d")
+     nil t ""]
+    ["\\.tbro?\\'"
+     "brotli compressing" "brotli" ("-c" "-w" "24" "--best")
+     "brotli uncompressing" "brotli" ("-c" "-d")
+     nil t ""]
+    ["\\.lz4\\'"
+     "lz4 compressing" "lz4" ("-c" "-q" "-9")
+     "lz4 uncompressing" "lz4" ("-c" "-q" "-d")
+     t t "\4\42\115\30"]
+    ["\\.tl4\\'"
+     "lz4 compressing" "lz4" ("-c" "-q" "-9")
+     "lz4 uncompressing" "lz4" ("-c" "-q" "-d")
+     t t "\004\042\115\030"]
+    ["\\.tl?z4\\'"
+     "lz4 compressing" "lz4" ("-c" "-q" "-9")
+     "lz4 uncompressing" "lz4" ("-c" "-q" "-d")
+     t t "\004\042\115\030"]
+    ["\\.zst\\'"
+     "zstd compressing" "zstd" ("-c" "-q" "-f" "-T0" "--ultra" "-22")
+     "zstd uncompressing" "zstd" ("-c" "-q" "-f" "-d" "--long=31")
+     t t "\050\265\057\375"]
+    ["\\.tzst?\\'"
+     "zstd compressing" "zstd" ("-c" "-q" "-f" "-T0" "--ultra" "-22")
+     "zstd uncompressing" "zstd" ("-c" "-q" "-f" "-d" "--long=31")
+     t t "\050\265\057\375"]
     ;; dzip is gzip with random access.  Its compression program can't
     ;; read/write stdin/out, so .dz files can only be viewed without
     ;; saving, having their contents decompressed with gzip.
@@ -308,7 +336,11 @@
 (defcustom jka-compr-mode-alist-additions
   (purecopy '(("\\.tgz\\'" . tar-mode)
               ("\\.tbz2?\\'" . tar-mode)
-              ("\\.txz\\'" . tar-mode)))
+              ("\\.txz\\'" . tar-mode)
+              ("\\.tbro?\\'" . tar-mode)
+              ("\\.tl4\\'" . tar-mode)
+              ("\\.tl?z4\\'" . tar-mode)
+              ("\\.tzst?\\'" . tar-mode)))
   "List of pairs added to `auto-mode-alist' when installing jka-compr.
 Uninstalling jka-compr removes all pairs from `auto-mode-alist' that
 installing added.
