--- 1/lib/portage/process.py
+++ 1/lib/portage/process.py
@@ -478,7 +478,8 @@
 
 	try:
 		subprocess.call(['ip', 'address', 'add', '10.0.0.1/8', 'dev', 'lo'])
-		subprocess.call(['ip', 'address', 'add', 'fd00::1/8', 'dev', 'lo'])
+		# This breaks on non-IPv6 systems:
+		#subprocess.call(['ip', 'address', 'add', 'fd00::1/8', 'dev', 'lo'])
 	except OSError as e:
 		writemsg("Error calling 'ip': %s\n" % e.strerror, noiselevel=-1)
 
