--- 1/bin/ecompress
+++ 1/bin/ecompress
@@ -71,9 +71,9 @@
 				# note: to save time, we need to do this only if there's
 				# at least one compressed file
 				case ${path} in
-					*.Z|*.gz|*.bz2|*.lzma|*.xz)
+					*.Z|*.gz|*.bz2|*.lzma|*.xz|*.br|*.lz4|*.zst)
 						vpath=${path%.*}
-						for comp in '' .Z .gz .bz2 .lzma .xz; do
+						for comp in '' .Z .gz .bz2 .lzma .xz .br .lz4 .zst; do
 							if [[ ${vpath}${comp} != ${path} && \
 									-e ${vpath}${comp} ]]; then
 								collisions[${path}]=1
--- 1/bin/ecompress-file
+++ 1/bin/ecompress-file
@@ -32,6 +32,15 @@
 			*.lz)
 				lzip -df "${x}" || __helpers_die "lzip -d failed"
 				x=${x%.lz};;
+			*.br)
+				brotli -dfj "${x}" || __helpers_die "brotli -dj failed"
+				x=${x%.*};;
+			*.lz4)
+				unlz4 -f -q --rm "${x}" || __helpers_die "unlz4 failed"
+				x=${x%.*};;
+			*.zst)
+				unzstd -f -q --long=31 --rm "${x}" || __helpers_die "unzstd failed"
+				x=${x%.*};;
 		esac
 
 		filtered_args+=( "$x" )
