--- 1/lib/portage/sync/modules/git/git.py
+++ 1/lib/portage/sync/modules/git/git.py
@@ -79,6 +79,13 @@
             )
             self.spawn_kwargs["env"].update(clone_env)
 
+        update_index_cmd = [self.bin_command,
+            'update-index', '--refresh', '-q', '--unmerged']
+        with open(os.devnull, 'wb') as devnull:
+            subprocess.call(update_index_cmd, stdout=devnull,
+                cwd=portage._unicode_encode(self.repo.location)
+            )
+
         if self.settings.get("PORTAGE_QUIET") == "1":
             git_cmd_opts += " --quiet"
         if self.repo.clone_depth is not None:
