From https://www.spinics.net/lists/util-linux-ng/msg16996.html
--- a/lib/pty-session.c
+++ b/lib/pty-session.c
@@ -292,7 +292,20 @@
 
 static int write_to_child(struct ul_pty *pty, char *buf, size_t bufsz)
 {
-	return write_all(pty->master, buf, bufsz);
+	sigset_t set, org;
+	int rc;
+
+	sigemptyset(&set);
+	sigemptyset(&org);
+	sigaddset(&set, SIGINT);
+	sigaddset(&set, SIGTERM);
+
+	sigprocmask(SIG_UNBLOCK, &set, &org);
+
+	rc = write_all(pty->master, buf, bufsz);
+
+	sigprocmask(SIG_SETMASK, &org, NULL);
+       return rc;
 }
 
 /*
